apps/_example05app/pages/auth/forgot.jsx

```diff
@@ -1,23 +1,129 @@
 /** @jsx jsx */
 import { css, jsx } from '@emotion/core'
-import { Typography } from 'antd'
+import { useState } from 'react'
+import { Button, Form, Input, notification, Typography, Result } from 'antd'
 import Head from 'next/head'
+import Router from 'next/router'
 import { useIntl } from 'react-intl'
+import gql from 'graphql-tag'
+import { useMutation } from '@apollo/react-hooks'
 
 import BaseLayout from '../../containers/BaseLayout'
+import qs from 'qs'
 
 const { Title, Paragraph } = Typography
 
-const SignInPage = () => {
+const START_PASSWORD_RECOVERY_MUTATION = gql`
+    mutation startPasswordRecovery($email: String!){
+        status: startPasswordRecovery(email: $email)
+    }
+`
+
+const layout = {
+    labelCol: { span: 8 },
+    wrapperCol: { span: 16 },
+}
+
+const tailLayout = {
+    wrapperCol: { offset: 8, span: 16 },
+}
+
+const ForgotForm = () => {
+    const [form] = Form.useForm()
+    const intl = useIntl()
+    const [isLoading, setIsLoading] = useState(false)
+    const [isSuccessMessage, setIsSuccessMessage] = useState(false)
+    const [startPasswordRecovery, ctx] = useMutation(START_PASSWORD_RECOVERY_MUTATION)
+    let initialValues = (typeof window !== 'undefined' && window.location && window.location.href.includes('?') ? qs.parse(window.location.href.split('?', 2)[1]) : {})
```

Comment:
We have duplicated code here, maybe we can create a function and move it away to something like `/utils`?

Reply:
Ok: extracted with tests
https://github.com/8iq/nodejs-hackathon-boilerplate-starter-kit/pull/1/commits/4c2243e782092e57d7658584621f20dfdb0d11a7